### ISSUE TITLE
Add .floxrc or something to add default options to environments

### ISSUE THREAD

**Issue by dg-nvm** [flox/flox#1829]
**Describe the feature:**

When I create environment it automatically gets cuda enabled. That adds 10s to startup (find scan over 26k files) and is not needed to run `jq` for example. I would like to have ~/.floxrc to globally set options so new environments would have them set as intended.

**Describe a use case of this feature:**

Add cuda=disabled option to ~/.floxrc so each new environment created has it by default

**Acceptance criteria:**

1. ~/.floxrc or ~/.config/.floxrc file exists and is a source of global defaults
2. creating new env with `flox init` will use these defaults and use options from them
3. changing / editing environment would not use defaults again

**Comment by mkenigs** [comment]
> When I create environment it automatically gets cuda enabled. That adds 10s to startup (find scan over 26k files)

@tomberek could we add a `maxdepth` to the find, or does it have to be recursive?
https://github.com/flox/flox/blob/3fdbf1d0dbc4158790b831c6fd089ecb5353b366/assets/activation-scripts/etc/profile.d/0800_cuda.sh#L25-L34

**Comment by dg-nvm** [comment]
@mkenigs ldconfig is available in iconv package and it can be used to scan outside world:

```
λ which ldconfig
/home/dgoslawski/t/iconv/.flox/run/x86_64-linux.iconv/bin/ldconfig
λ ldconfig -C /etc/ld.so.cache -p -n | grep libcuda.so.1 | awk -F'=> ' '{print $2}'
/usr/lib/wsl/lib/libcuda.so.1
```

It is faster, and if it fails it can then go back to old method

**Comment by tomberek** [comment]
Yep, both approqches are viable to cut down on the cost of activation.

**Comment by dg-nvm** [comment]
on my machine maxdepth needed to traverse all is 12, depth needed to detect file (/usr/lib/wsl/lib/libcuda.so.1) is 3.

**Comment by mkenigs** [comment]
I split the performance of the hook out into a separate issue https://github.com/flox/flox/issues/1835

We can use this one for the feature request of supporting default options
